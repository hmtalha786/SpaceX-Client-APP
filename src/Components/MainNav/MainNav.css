.main-nav{
    text-align: center;
    background-color: black;
    width: 100%;
    justify-content: center;
    display: flex;
}
.nav-link{
    cursor: pointer;
    text-decoration: none;
    color:white;
    display: block;
    font-family: 'Sawarabi Gothic', sans-serif;
    position: relative;
    margin-left: 25px;
    margin-right: 25px;
    transition: color 0.5s ;
    transition-timing-function: ease-in-out;
}

.nav-link:hover{
    color: rgb(145, 143, 143);
}

@media only screen and (max-width: 450px){
    .main-nav{
        display: block;
    }
    p{
        margin: 0;
    }
}