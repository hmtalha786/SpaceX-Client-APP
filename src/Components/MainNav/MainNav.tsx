import React from 'react'
import './MainNav.css'
import { Link } from 'react-router-dom'

const MainNav = () => {
    return (
        <div className="main-nav">
            <Link className="nav-link" to="/" ><p>Home</p></Link>
            <Link className="nav-link" to="/launches"><p>Launches</p></Link>
            <Link className="nav-link" to="/rockets"><p>Rockets</p></Link>
            <Link className="nav-link" to="/ships"><p>Ships</p></Link>
            
        </div>
    )
}

export default MainNav
