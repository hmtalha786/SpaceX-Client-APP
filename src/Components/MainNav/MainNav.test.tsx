import React from 'react';
import {shallow} from 'enzyme';
import MainNav from './MainNav';

describe('renders nav' , () => {
    let container : any;

    beforeEach(() => container = shallow(<MainNav />))

    it('renders the main nav' , () => {
        expect(container).toMatchSnapshot()
    })

    it('renders the div' , () => {
        expect(container.find('div').length).toEqual(1)
    })
})