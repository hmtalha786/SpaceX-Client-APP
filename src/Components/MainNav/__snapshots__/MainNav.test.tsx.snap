// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders nav renders the main nav 1`] = `
<div
  className="main-nav"
>
  <Link
    className="nav-link"
    to="/"
  >
    <p>
      Home
    </p>
  </Link>
  <Link
    className="nav-link"
    to="/launches"
  >
    <p>
      Launches
    </p>
  </Link>
  <Link
    className="nav-link"
    to="/rockets"
  >
    <p>
      Rockets
    </p>
  </Link>
  <Link
    className="nav-link"
    to="/ships"
  >
    <p>
      Ships
    </p>
  </Link>
</div>
`;
